cpp/src/arrow/compute/exec/options.h

```diff
@@ -214,9 +214,11 @@ struct ARROW_EXPORT BackpressureOptions {
 class ARROW_EXPORT SinkNodeOptions : public ExecNodeOptions {
  public:
   explicit SinkNodeOptions(std::function<Future<std::optional<ExecBatch>>()>* generator,
```

Comment:
Do we want to ensure backwards compatibility and still provide the former 3-argument constructor?

Reply:
Good idea.  Done.